src/main/java/betteradvancements/util/CriterionGrid.java

```diff
@@ -0,0 +1,132 @@
+package betteradvancements.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import net.minecraft.advancements.Advancement;
+import net.minecraft.advancements.AdvancementProgress;
+import net.minecraft.advancements.Criterion;
+import net.minecraft.client.gui.FontRenderer;
+
+public class CriterionGrid {
+    public static boolean showSpoilers = false;
+
+    private final List<String> cellContents;
+    private final int[] cellWidths;
+    private final int fontHeight;
+    private final int numColumns;
+    public final int numRows;
+    public List<Column> columns;
+    public int width;
+    public int height;
+    public double aspectRatio;
+
+    public CriterionGrid() {

```

Comment:
This constructor should be `private` as it is only used internally.

Reply:
Agreed